docs/data-model.rst

```diff
@@ -656,10 +656,9 @@ For simplicity and algorithmic efficiency, mutations must also:
   **before** their children (i.e. if a mutation with ID :math:`x` has
   ``parent`` with ID :math:`y`, then we must have :math:`y < x`).
 
-Violations of these sorting requirements are detected at load time.
-The :meth:`TableCollection.sort` method ensures that mutations are sorted
-according site ID, but does not at present enforce that mutations occur
-after their parent mutations.
+Violations of these sorting requirements are detected at load time. The
+:meth:`TableCollection.sort` method will ensure that these sortedness
```

Comment:
Well, not quite - what if all the mutation times are unknown?

Reply:
Doesn't `sort` then sort so that parents appear before children? I assumed this had been fixed too.

Reply:
No, it has no way of knowing. It's just stable in that it doesn't rearrange things.

Reply:
Oh, sorry, I though that had been fixed too. So https://github.com/tskit-dev/tskit/issues/27 is still an active issue I guess. Is there even a solution (as this is a bit annoying!)

Reply:
Perhaps this should say
```
but does not enforce that mutations occur after their parent mutations if mutation times are unknown.
```

If the `parent` attribute is correct then it could be used to ensure that the sortedness criteria are satisfied even without times. I think we didn't do this because we didn't have a use case where parent mutation IDs are known but out of order.

Reply:
So is there any way to change the sorting code so that the example I gave in https://github.com/tskit-dev/tskit/issues/27#issuecomment-562062932 actually works (i.e. sort in the correct way to allow `compute_mutation_parents` to do its stuff?

Reply:
(refreshing my memory of the discussion in #651...)  Well, if you put a mutation.time in that example, then the sorting code *will* do what you want. Otherwise, I that sorting mutations to respect ordering on the trees will require constructing the trees, which we'd really like to avoid doing. Do you have a use case that this is making things difficult for?

Reply:
I don't think we should change the sorting order @hyanwong - we went through it all in great detail the first time around and we ended up with a compromise that works well. There's no reason to change it now I think.